### PR TITLE
Create node-local-dns.md

### PR DESCRIPTION
**Pull request by changluyi**
- [ ] Make sure you have followed the [Document Convention](../docs/reference/document-convention.md).

### CODE REVIEW

File: docs/advance/node-local-dns.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# NodeLocal DNSCache 和 kube-ovn 适配
+
+NodeLocal DNSCache 是通过集群节点上作为 DaemonSet 运行 DNS 缓存来提高集群 DNS 性能，该功能也可以和 kube-ovn 适配。
+
+## 节点本地 DNS 缓存部署
+
+### 部署 k8s 的节点本地 DNS 缓存
+
+该步骤参考 k8s 官网配置 [nodelocaldnscache](https://kubernetes.io/zh-cn/docs/tasks/administer-cluster/nodelocaldns/)。
```
Comment (oilbeater) [inline docs/advance/node-local-dns.md:9]: k8s -> Kubernetes

File: docs/advance/node-local-dns.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# NodeLocal DNSCache 和 kube-ovn 适配
+
+NodeLocal DNSCache 是通过集群节点上作为 DaemonSet 运行 DNS 缓存来提高集群 DNS 性能，该功能也可以和 kube-ovn 适配。
```
Comment (oilbeater) [inline docs/advance/node-local-dns.md:3]: kube-ovn -> Kube-OVN

File: docs/advance/node-local-dns.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+```bash
+kubectl edit deployment kube-ovn-controller -n kube-system
+```
+
+给 spec.template.spec.containers.args 增加字段 --node-local-dns-ip=169.254.20.10
+
+#### 重建已经创建的 pod
+
+这步原因是让 pod 重新生成 /etc/resolv.conf 让 nameserver 指向本地 dns ip，如果没有重建 pod 的 nameserver 将仍然使用集群的 dns cluster ip。同时 u2o 开关如果开启也需要重建 pod 来重新生成 pod 网关。
```
Comment (oilbeater) [inline docs/advance/node-local-dns.md:46]:
```suggestion
这步原因是让 pod 重新生成 `/etc/resolv.conf` 让 nameserver 指向本地 DNS IP，如果没有重建 Pod 的 nameserver 将仍然使用集群的 DNS ClusterIP。同时 u2o 开关如果开启也需要重建 pod 来重新生成 pod 网关。
```

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: docs/advance/node-local-dns.md (hunk 3 above)
Comment (changluyi) [reply docs/advance/node-local-dns.md]: fixed all

File: docs/advance/node-local-dns.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+wget https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml
+sed -i "s/__PILLAR__LOCAL__DNS__/$localdns/g; s/__PILLAR__DNS__DOMAIN__/$domain/g; s/,__PILLAR__DNS__SERVER__//g; s/__PILLAR__CLUSTER__DNS__/$kubedns/g" nodelocaldns.yaml
+
+kubectl apply -f nodelocaldns.yaml
+```
+
+修改每个节点上的 kubelet 配置文件，将 /var/lib/kubelet/config.yaml 中的 clusterDNS 字段修改为本地 DNS IP 169.254.20.10，然后重启 kubelet 服务。
+
+### kube-OVN 相应 DNS 配置
```
Comment (oilbeater) [inline docs/advance/node-local-dns.md:28]:
```suggestion
### Kube-OVN 相应 DNS 配置
```

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: docs/advance/node-local-dns.md (hunk 4 above)
Comment (changluyi) [reply docs/advance/node-local-dns.md]: fixed

**Pull request merged**